src-tauri/src/mod_management/mod.rs

```diff
@@ -679,14 +679,22 @@ pub fn delete_northstar_mod(game_install: GameInstall, nsmod_name: String) -> Re
     }
 
     // Get installed mods
-    let installed_ns_mods = get_installed_mods_and_properties(game_install)?;
+    let installed_ns_mods = get_installed_mods_and_properties(game_install.clone())?;
 
     // Get folder name based on northstarmods
     for installed_ns_mod in installed_ns_mods {
         // Installed mod matches specified mod
         if installed_ns_mod.name == nsmod_name {
             // Delete folder
-            return delete_mod_folder(&installed_ns_mod.directory);
+            return match delete_thunderstore_mod(
+                game_install,
+                installed_ns_mod
+                    .thunderstore_mod_string
+                    .unwrap_or("".to_string()),
```

Comment:
Uh, I wouldn't pass an empty string (on `None`) to `delete_thunderstore_mod()` cause that will cause it to not match on anything and then cause the legacy version which will panic here:

https://github.com/R2NorthstarTools/FlightCore/blob/264872e0b1d6aa5572fc2f9504d73845505ce973/src-tauri/src/mod_management/legacy.rs#L170-L171

Instead check if `thunderstore_mod_string` is `Some()` and then try to attempt deleting that mod.